### ISSUE TITLE
Defects found in Accessibility Check - continue for #1353

### ISSUE THREAD

**Issue by MiraYuan** [IBM/carbon-components-angular#1360]
## Detailed description

> Describe in detail the issue you're having.
We need do AVT(Accessibility Verification Test) for our project, I found 10 issues(7 of them are below to 1 type problem, 2 of them are belong 1 type problem) cannot be solved from my side. And I opened #1353 ,with your solution to update version to 3.51.0, issues about \<ibm-table> are solved. However, a issue about \<ibm-checkbox> are still remained.
Please help fixing it.

You can download the IBM Equal Access Accessibility Checker from website
"https://www.ibm.com/able/toolkit/tools#verification-tools"
to install the Accessibility Checker. If you are using Chrome, for example, after the installation, open the developer tool, and select 'Accessibility Checker'.

Element location:
\<ibm-checkbox>
          \<input>\</input>
\</ibm-checkbox>
problem: 
Checkbox input is not in the group with another Checkbox with the name "undefined"
Related sets of radio buttons or checkboxes should be programmatically grouped
IBM 1.3.1 Info and relationships
What to do:
        Group related controls of the same type by enclosing them within a <fieldset> element;
    OR, group related radio buttons by using role="radiogroup";
    OR, group related checkboxes by using role="group";
    OR, disassociate unrelated controls or controls of different types by using different "name" attribute values.

> Is this a feature request (new component, new icon), a bug, or a general issue?
a bug
> Is this issue related to a specific component?
no
> What did you expect to happen? What happened instead? What would you like to see changed?
Fixed them so that our project could pass AVT.
> What browser are you working in?
Firefox
> What version of the Carbon Design System are you using?
^3.51.0
> What offering/product do you work on? Any pressing ship or release dates we should be aware of?
4Q of 2020
## Steps to reproduce the issue

1. Step one: Download the IBM Equal Access Accessibility Checker for AVT. (https://www.ibm.com/able/toolkit/tools#verification-tools)
2. Step two: Scan page with tabs mentioned before.
3. After scan(very quickly), it's obvious to see the problems.

## Additional information

* Screenshots or code
* Notes

**Comment by MiraYuan** [comment]
how can I get this fix?

**Comment by Donisius** [comment]
@MiraYuan Hi, should be available in version 3.53.2

**Comment by MiraYuan** [comment]
I have tried just now, it seems not work. Not only this not can fix the problem, the fix for <ibm-table> in 3.51.0 is not working too...

**Comment by MiraYuan** [comment]
Who does this affect?

    Blind people using screen readers who cannot see the context around the control
    People with dexterity impairment using voice control

Why is this important?

Associating a meaningful label with every UI control allows the browser and assistive technology to expose and announce the control to a user. Associating a visible label also provides a larger clickable area.
Violation
Form control element <input> has no associated label
Each form control must have an associated label

IBM 3.3.2 Labels or Instructions | WCAG 2.1 technique H44 | WCAG 2.1 technique ARIA16
Element location

<input aria-checked="true" id="checkbox-4" type="checkbox" class="bx--checkbox">

What to do

    Add a for attribute to a visible label with the id of this control (e.g. <label for="controlID">);
    OR put the UI element inside a visible <label> element;
    OR add an aria-labelledby attribute to the control. It must point to visible text on the page that is meaningful as a label;
    OR only if the design cannot have a visible label, use the title attribute to provide a label.

**Comment by Donisius** [comment]
@MiraYuan If you put the checkboxes into a `fieldset` like in https://github.com/IBM/carbon-components-angular/blob/master/src/checkbox/checkbox.stories.ts#L16-L38 it seems to also work

**Comment by MiraYuan** [comment]
Hi, I put the checkboxes into a fieldset like you mentioned. There are three problems happen for ibm-checkbox and one problem happen in ibm-checkbox under ibm-table.
1.

Who does this affect?

    Blind people using screen readers
    People with dexterity impairment using voice control

Why is this important?

Assistive technologies use labels to identify form controls. Providing more than one label for the same form control causes confusion, and may be handled differently by different browsers.
Violation
Form control has more than one label
Form controls should have exactly one label

IBM 1.3.1 Info and Relationships | WCAG 2.1 technique H44
Element location

<label aria-label="Select row" for="checkbox-10" class="bx--checkbox-label">

What to do

    If the for attribute references a form control, find all <label> elements with this for attribute value. Decide which is the intended label and remove or change the others;
    OR, if the for attribute does not reference a form control, remove the label or the for attribute.

2.

Who does this affect?

    Blind people using screen readers
    People with dexterity impairment using voice control

Why is this important?

The <legend> of a <fieldset> element provides a label or instruction for the group of controls within the <fieldset>. This label helps users to recognize that the controls are related. Controls grouping is most important for related radio buttons and checkboxes, but can also be useful for other sets of controls that are not as tightly related.
Violation
<fieldset> element does not have a <legend>
<fieldset> elements must have a single, non-empty <legend> as a label

IBM 1.3.1 Info and Relationships | WCAG 2.1 technique H71
Element location

<fieldset class="bx--fieldset" _ngcontent-axy-c5="">

What to do

    Place a single <legend> as the first element in the <fieldset> group. The legend must contain a brief description of the related fields enclosed by the <fieldset>. For example:

<fieldset>
       <legend>Skill Level</legend>
       <input type="radio" name="skillLevel" value="high" id="highLev">
       <label for="highLev">High</label><br>
       <input type="radio" name="skillLevel" value="novice" id="noviceLev">
       <label for="noviceLev">Novice</label>
</fieldset>

3. 

Who does this affect?

    Blind people using screen readers
    People with dexterity impairment using voice control

Why is this important?

Assistive technologies use labels to identify form controls. Providing more than one label for the same form control causes confusion, and may be handled differently by different browsers.
Violation
Form control has more than one label
Form controls should have exactly one label

IBM 1.3.1 Info and Relationships | WCAG 2.1 technique H44
Element location

<label aria-label="" for="checkbox-3" class="bx--checkbox-label">

What to do

    If the for attribute references a form control, find all <label> elements with this for attribute value. Decide which is the intended label and remove or change the others;
    OR, if the for attribute does not reference a form control, remove the label or the for attribute.


Problems of checkbox in <ibm-table>
1. 

Who does this affect?

    Blind people using screen readers
    People with dexterity impairment using voice control

Why is this important?

Assistive technologies use labels to identify form controls. Providing more than one label for the same form control causes confusion, and may be handled differently by different browsers.
Violation
Form control has more than one label
Form controls should have exactly one label

IBM 1.3.1 Info and Relationships | WCAG 2.1 technique H44
Element location

<label aria-label="Select all rows" for="checkbox-6" class="bx--checkbox-label">

What to do

    If the for attribute references a form control, find all <label> elements with this for attribute value. Decide which is the intended label and remove or change the others;
    OR, if the for attribute does not reference a form control, remove the label or the for attribute.

**Comment by MiraYuan** [comment]
It works well. Thanks for your quickly response!
